syncd/syncd.cpp

```diff
@@ -2936,6 +2936,88 @@ bool handleRestartQuery(swss::NotificationConsumer &restartQuery)
     return false;
 }
 
+bool getSwitchRid(sai_object_id_t* switch_rid)
```

Comment:
switch rid can be obtained in much easier way: global object "switches" contains all the switches where key is VID of the switch, take a look syncd.cpp at line 50, usually there will be 1 or 0 switches

this is not needed

Reply:
optimized as suggested.